__init__.py

```diff
@@ -11,1012 +11,863 @@
+            timers filtered based on the answer to the question.
         """
+        filtered_timers = None
+        speakable_matches = self._get_speakable_timer_details(timers)
+        reply = self.get_response(
+            dialog=question, data=dict(count=len(timers), names=speakable_matches)
+        )
+        if reply is not None:
+            filtered_timers = get_timers_matching_reply(
+                reply, timers, self.regex_file_path
+            )
 
-        with self.file_system.open(name, 'wb') as f:
-            pickle.dump(data, f, pickle.HIGHEST_PROTOCOL)
+        return filtered_timers
 
-    def do_unpickle(self, name, default):
-        """Load previously saved data under name.
+    def _get_speakable_timer_details(self, timers: List[CountdownTimer]) -> str:
+        """Get timer list as speakable string.
 
         Args:
-            name (string): reference name of the pickled data
-            default (any): default if data isn't found
+            timers: the timers to be converted
 
         Returns:
-            (any): Picked data or the default
+            names of the specified timers to be passed to TTS engine for speaking
         """
-        try:
-            with self.file_system.open(name, 'rb') as f:
-                return pickle.load(f)
-        except:
-            return default
+        speakable_timer_details = []
+        for timer in timers:
+            dialog = TimerDialog(timer, self.lang)
+            dialog.build_details_dialog()
+            speakable_timer_details.append(self.translate(dialog.name, dialog.data))
+        timer_names = join_list(speakable_timer_details, self.translate("and"))
+
+        return timer_names
+
+    def handle_listener_started(self, _):
+        """Pause scheduled events that would interfere with the device listening."""
+        self._pause_scheduled_events()
+
+    def handle_listener_ended(self, _):
+        """Resume scheduled events paused when the listener started."""
+        self._resume_scheduled_events()
+
+    def handle_speak(self, _):
+        """Pause scheduled events while the device speaking."""
+        self._pause_scheduled_events()
+        wait_while_speaking()
+        self._resume_scheduled_events()
+
+    def _pause_scheduled_events(self):
+        """Pause scheduled events that interfere with device operations.
+
+        All devices will be beeping if there are one or more expired timers.  Pause
+        the beeping so that the device can listen and respond to a user request.
+
+        The Mark I has mouth events that occur during listening and speaking.  Stop
+        displaying the timer during these events.
+        """
+        self._stop_expiration_check()
+        if self.platform == MARK_I:
+            self._stop_timer_display()
+
+    def _resume_scheduled_events(self):
+        """Resume scheduled events that were paused during listening/speaking."""
+        self._start_timer_display()
+        if self.platform == MARK_I:
```

Comment:
Should this line be
```python
        if len(self.active_timers > 0):
```

Was chatting with @emphasize about this and the timer events weren't being triggered properly. It turned out it was stopped and then not restarted. When we debugged we thought this looked a bit weird, and making the above change makes the event trigger as  expected.

Reply:
Just pushed a few changes regarding the timing of some Mark I faceplate logic.  I think it will address your concerns.